src/main/java/io/github/programminglife2016/pl1_2016/parser/TreeNode.java

```diff
@@ -0,0 +1,114 @@
+package io.github.programminglife2016.pl1_2016.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class for representing a node in the phylogenetic tree.
+ */
+public class TreeNode implements JsonSerializable {
```

Comment:
Wouldn't it be better if `TreeNode` was an interface? `JsonSerializable` is not specific as an interface.